### PR TITLE
Add GRS

### PR DESCRIPTION
**Pull request by HashEngineering**

### CODE REVIEW

File: bitcoinlib/data/networks.json [hunk 1]
```diff
@@ -246,5 +246,71 @@
     "fee_min": 1000,
     "fee_max": 50000,
     "priority": 6
+  },
+  "groestlcoin":
+  {
+    "description": "Groestlcoin",
+    "currency_name": "groestlcoin",
+    "currency_name_plural": "Groestlcoins",
```
Comment (gruve-p) [inline bitcoinlib/data/networks.json]: "currency_name_plural": "groestlcoins",

File: bitcoinlib/data/providers.examples.json [hunk 2]
```diff
@@ -394,5 +394,93 @@
     "priority": 10,
     "denominator": 100000000,
     "network_overrides": null
+  },
+  "groestlcoind": {
+    "provider": "groestlcoind",
+    "network": "bitcoin",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "network": "groestlcoin",

File: bitcoinlib/data/providers.examples.json [hunk 3]
```diff
@@ -394,5 +394,93 @@
...
+    "provider_coin_id": "",
+    "url": "",
+    "api_key": "",
+    "priority": 20,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groestlcoind.testnet": {
+    "provider": "groestlcoind",
+    "network": "testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.examples.json [hunk 4]
```diff
@@ -394,5 +394,93 @@
...
+    "client_class": "GroestlcoindClient",
+    "provider_coin_id": "",
+    "url": "",
+    "api_key": "",
+    "priority": 20,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora": {
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "groestlcoinesplora": {

File: bitcoinlib/data/providers.examples.json [hunk 5]
```diff
@@ -394,5 +394,93 @@
...
+    "provider": "groesltcoinesplora",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "provider": "groestlcoinesplora",

File: bitcoinlib/data/providers.examples.json [hunk 6]
```diff
@@ -394,5 +394,93 @@
...
+    "network": "groestlcoin",
+    "client_class": "GroestlcoinEsploraClient",
+    "provider_coin_id": "",
+    "url": "https://esplora.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora.testnet": {
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "groestlcoinesplora.testnet": {

File: bitcoinlib/data/providers.examples.json [hunk 7]
```diff
@@ -394,5 +394,93 @@
...
+    "provider": "groesltcoinesplora",
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.examples.json [hunk 8]
```diff
@@ -394,5 +394,93 @@
...
+    "client_class": "GroestlcoinEsploraClient",
+    "provider_coin_id": "",
+    "url": "https://esplora.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora.testnet": {
+    "provider": "groesltcoinesplora",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "provider": "groestlcoinesplora",

File: bitcoinlib/data/providers.examples.json [hunk 9]
```diff
@@ -394,5 +394,93 @@
...
+    "provider_coin_id": "",
+    "url": "https://groestlsight.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groestlsight.testnet": {
+    "provider": "groestlsight",
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.examples.json [hunk 10]
```diff
@@ -394,5 +394,93 @@
...
+    "provider_coin_id": "grs",
+    "url": "https://chainz.cryptoid.info/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "cryptoid.groestlcoin.testnet": {
+    "provider": "cryptoid",
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.examples.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.json [hunk 11]
```diff
@@ -306,5 +306,93 @@
...
+  "groestlcoind.testnet": {
+    "provider": "groestlcoind",
+    "network": "testnet",
+    "client_class": "GroestlcoindClient",
+    "provider_coin_id": "",
+    "url": "",
+    "api_key": "",
+    "priority": 20,
+    "denominator": 100000000,
+    "network_overrides": null
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: Not sure if these should be included in the providers.json

File: bitcoinlib/data/providers.json [hunk 12]
```diff
@@ -306,5 +306,93 @@
...
+  },
+  "groesltcoinesplora": {
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "groestlcoinesplora": {

File: bitcoinlib/data/providers.json [hunk 13]
```diff
@@ -306,5 +306,93 @@
...
+    "provider": "groesltcoinesplora",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "provider": "groestlcoinesplora",

File: bitcoinlib/data/providers.json [hunk 14]
```diff
@@ -306,5 +306,93 @@
...
+    "network": "groestlcoin",
+    "client_class": "BlockstreamClient",
+    "provider_coin_id": "",
+    "url": "https://esplora.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora.testnet": {
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "groestlcoinesplora.testnet": {

File: bitcoinlib/data/providers.json [hunk 15]
```diff
@@ -306,5 +306,93 @@
...
+    "provider": "groesltcoinesplora",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "provider": "groestlcoinesplora",

File: bitcoinlib/data/providers.json [hunk 16]
```diff
@@ -306,5 +306,93 @@
...
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.json [hunk 17]
```diff
@@ -306,5 +306,93 @@
...
+    "provider_coin_id": "",
+    "url": "https://groestlsight.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groestlsight.testnet": {
+    "provider": "groestlsight",
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.json [hunk 18]
```diff
@@ -306,5 +306,93 @@
...
+    "provider_coin_id": "grs",
+    "url": "https://chainz.cryptoid.info/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "cryptoid.groestlcoin.testnet": {
+    "provider": "cryptoid",
+    "network": "groestlcoin.testnet",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "network": "groestlcoin_testnet",

File: bitcoinlib/data/providers.json [hunk 19]
```diff
@@ -306,5 +306,93 @@
...
+    "url": "https://esplora.groestlcoin.org/testnet/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groestlsight": {
+    "provider": "groestlsight",
+    "network": "groestlcoin",
+    "client_class": "InsightDashClient",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "client_class": "GroestlsightClient",

File: bitcoinlib/data/providers.json [hunk 20]
```diff
@@ -306,5 +306,93 @@
...
+    "url": "https://groestlsight.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groestlsight.testnet": {
+    "provider": "groestlsight",
+    "network": "groestlcoin.testnet",
+    "client_class": "InsightDashClient",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "client_class": "GroestlsightClient",

File: bitcoinlib/data/providers.json [hunk 21]
```diff
@@ -306,5 +306,93 @@
...
+    "url": "",
+    "api_key": "",
+    "priority": 20,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora": {
+    "provider": "groesltcoinesplora",
+    "network": "groestlcoin",
+    "client_class": "BlockstreamClient",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "client_class": "GroestlcoinEsploraClient",

File: bitcoinlib/data/providers.json [hunk 22]
```diff
@@ -306,5 +306,93 @@
...
+    "url": "https://esplora.groestlcoin.org/api/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
+  },
+  "groesltcoinesplora.testnet": {
+    "provider": "groesltcoinesplora",
+    "network": "groestlcoin.testnet",
+    "client_class": "BlockstreamClient",
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json]: "client_class": "GroestlcoinEsploraClient",

File: bitcoinlib/services/groestlcoind.py [hunk 23]
```diff
@@ -0,0 +1,271 @@
...
+def _read_from_config(configparser, section, value, fallback=None):
+    try:
+        return configparser.get(section, value)
+    except Exception:
+        return fallback
+
+
+class GroestlcoindClient(BaseClient):
+    """
+    Class to interact with groestlcoind, the Bitcoin deamon
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: Class to interact with groestlcoind, the Groestlcoin deamon

File: bitcoinlib/services/groestlcoind.py [hunk 24]
```diff
@@ -0,0 +1,271 @@
...
+    """
+
+    @staticmethod
+    def from_config(configfile=None, network='bitcoin'):
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: def from_config(configfile=None, network='groestlcoin'):

File: bitcoinlib/services/groestlcoind.py [hunk 25]
```diff
@@ -0,0 +1,271 @@
...
+        """
+        Read settings from groestlcoind config file
+
+        :param configfile: Path to config file. Leave empty to look in default places
+        :type: str
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: :param network: Groestlcoin mainnet or testnet. Default is groestlcoin mainnet

File: bitcoinlib/services/groestlcoind.py [hunk 26]
```diff
@@ -0,0 +1,271 @@
...
+        config_fn = 'bitcoin.conf'
+        if isinstance(network, Network):
+            network = network.name
+        if network == 'testnet':
+            config_fn = 'bitcoin-testnet.conf'
+
+        cfn = None
+        if not configfile:
+            config_locations = ['~/.bitcoinlib', '~/.bitcoin', '~/Application Data/Bitcoin',
+                                '~/Library/Application Support/Bitcoin']
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: config_locations = ['~/.bitcoinlib', '~/.groestlcoin', '~/Application Data/Groestlcoin',
                                '~/Library/Application Support/Groestlcoin']

File: bitcoinlib/services/groestlcoind.py [hunk 27]
```diff
@@ -0,0 +1,271 @@
...
+            for location in config_locations:
+                cfn = Path(location, config_fn).expanduser()
+                if cfn.exists():
+                    break
+        else:
+            cfn = Path(BCL_DATA_DIR, 'config', configfile)
+        if not cfn or not cfn.is_file():
+            raise ConfigError("Config file %s not found. Please install bitcoin client and specify a path to config "
+                              "file if path is not default. Or place a config file in .bitcoinlib/bitcoin.conf to "
+                              "reference to an external server." % cfn)
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: raise ConfigError("Config file %s not found. Please install groestlcoin client and specify a path to config "
                              "file if path is not default. Or place a config file in .bitcoinlib/groestlcoin.conf to "
                              "reference to an external server." % cfn)

File: bitcoinlib/services/groestlcoind.py [hunk 28]
```diff
@@ -0,0 +1,271 @@
...
+
+        testnet = _read_from_config(config, 'rpc', 'testnet')
+        if testnet:
+            network = 'testnet'
+        if _read_from_config(config, 'rpc', 'rpcpassword') == 'specify_rpc_password':
+            raise ConfigError("Please update config settings in %s" % cfn)
+        if network == 'testnet':
+            port = 18332
+        else:
+            port = 8332
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: port = 1441

File: bitcoinlib/services/groestlcoind.py [hunk 29]
```diff
@@ -0,0 +1,271 @@
...
+                config_string = '[rpc]\n' + f.read()
+            config.read_string(config_string)
+
+        testnet = _read_from_config(config, 'rpc', 'testnet')
+        if testnet:
+            network = 'testnet'
+        if _read_from_config(config, 'rpc', 'rpcpassword') == 'specify_rpc_password':
+            raise ConfigError("Please update config settings in %s" % cfn)
+        if network == 'testnet':
+            port = 18332
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: port = 17766

File: bitcoinlib/services/groestlcoind.py [hunk 30]
```diff
@@ -0,0 +1,271 @@
...
+        port = _read_from_config(config, 'rpc', 'rpcport', port)
+        server = '127.0.0.1'
+        server = _read_from_config(config, 'rpc', 'rpcconnect', server)
+        server = _read_from_config(config, 'rpc', 'bind', server)
+        server = _read_from_config(config, 'rpc', 'externalip', server)
+
+        url = "http://%s:%s@%s:%s" % (config.get('rpc', 'rpcuser'), config.get('rpc', 'rpcpassword'), server, port)
+        return GroestlcoindClient(network, url)
+
+    def __init__(self, network='bitcoin', base_url='', denominator=100000000, *args):
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: `def __init__(self, network='groestlcoin', base_url='', denominator=100000000, *args):`

File: bitcoinlib/services/groestlcoind.py [hunk 31]
```diff
@@ -0,0 +1,271 @@
...
+        """
+        try:
+            config = configparser.ConfigParser(strict=False)
+        except TypeError:
+            config = configparser.ConfigParser()
+        config_fn = 'bitcoin.conf'
+        if isinstance(network, Network):
+            network = network.name
+        if network == 'testnet':
+            config_fn = 'bitcoin-testnet.conf'
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: config_fn = 'groestlcoin-testnet.conf'

File: bitcoinlib/services/groestlcoind.py [hunk 32]
```diff
@@ -0,0 +1,271 @@
...
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
+        :type: str
+
+        :return GroestlcoindClient:
+        """
+        try:
+            config = configparser.ConfigParser(strict=False)
+        except TypeError:
+            config = configparser.ConfigParser()
+        config_fn = 'bitcoin.conf'
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: config_fn = 'groestlcoin.conf'

File: bitcoinlib/services/groestlcoind.py [hunk 33]
```diff
@@ -0,0 +1,271 @@
...
+        server = _read_from_config(config, 'rpc', 'rpcconnect', server)
+        server = _read_from_config(config, 'rpc', 'bind', server)
+        server = _read_from_config(config, 'rpc', 'externalip', server)
+
+        url = "http://%s:%s@%s:%s" % (config.get('rpc', 'rpcuser'), config.get('rpc', 'rpcpassword'), server, port)
+        return GroestlcoindClient(network, url)
+
+    def __init__(self, network='bitcoin', base_url='', denominator=100000000, *args):
+        """
+        Open connection to bitcoin node
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: Open connection to groestlcoin node

File: bitcoinlib/services/groestlcoind.py [hunk 34]
```diff
@@ -0,0 +1,271 @@
...
+
+        :param network: Bitcoin mainnet or testnet. Default is bitcoin mainnet
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: :param network: Groestlcoin mainnet or testnet. Default is groestlcoin mainnet

File: bitcoinlib/services/groestlcoind.py [hunk 35]
```diff
@@ -0,0 +1,271 @@
...
+        :type: str
+        :param base_url: Connection URL in format http(s)://user:password@host:port.
+        :type: str
+        :param denominator: Denominator for this currency. Should be always 100000000 (satoshis) for bitcoin
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: :param denominator: Denominator for this currency. Should be always 100000000 (gros) for groestlcoin

File: bitcoinlib/services/groestlcoind.py [hunk 36]
```diff
@@ -0,0 +1,271 @@
...
+        """
+        if isinstance(network, Network):
+            network = network.name
+        if not base_url:
+            bdc = self.from_config('', network)
+            base_url = bdc.base_url
+            network = bdc.network
+        if len(base_url.split(':')) != 4:
+            raise ConfigError("Groestlcoind connection URL must be of format 'http(s)://user:password@host:port,"
+                              "current format is %s. Please set url in providers.json file or check bitcoin config "
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: "current format is %s. Please set url in providers.json file or check groestlcoin config "

File: bitcoinlib/services/groestlcoind.py [hunk 37]
```diff
@@ -0,0 +1,271 @@
...
+                              "file" % base_url)
+        if 'password' in base_url:
+            raise ConfigError("Invalid password in groestlcoind provider settings. "
+                              "Please replace default password and set url in providers.json or bitcoin.conf file")
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: "Please replace default password and set url in providers.json or groestlcoin.conf file")

File: bitcoinlib/services/groestlcoind.py [hunk 38]
```diff
@@ -0,0 +1,271 @@
...
+
+if __name__ == '__main__':
+    #
+    # SOME EXAMPLES
+    #
+
+    from pprint import pprint
+
+    # 1. Connect by specifying connection URL
+    # base_url = 'http://bitcoinrpc:passwd@host:8332'
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py]: `# base_url = 'http://RpcUsername:RpcPass@host:1441`

**Comment by gruve-p** [comment]
Services for groestlsight (you can copy from insightdash.py) and groestlcoinesplora (copy from blockstream.py) should be added in https://github.com/Groestlcoin/btcrecover/tree/master/bitcoinlib/services. Maybe also add blockchair into https://github.com/Groestlcoin/btcrecover/blob/master/bitcoinlib/data/providers.examples.json and https://github.com/Groestlcoin/btcrecover/blob/master/bitcoinlib/data/providers.json for now

**Review by gruve-p — CHANGES_REQUESTED** [review]
Services for groestlsight (you can copy from bitcoinlib/services/insightdash.py) and groestlcoinesplora (copy from bitcoinlib/services/blockstream.py) should be added in bitcoinlib/services/. Maybe also add blockchair into bitcoinlib/data/providers.examples.json and bitcoinlib/data/providers.json for now. Is this implementation tested? Is importing groestlcoin_hash not needed for base58 checksum? Also we need to add unit tests for GRS on bitcoinlib/tests

**Comment by Stouse49** [comment]
> Services for groestlsight (you can copy from bitcoinlib/services/insightdash.py) and groestlcoinesplora (copy from bitcoinlib/services/blockstream.py) should be added in bitcoinlib/services/. Maybe also add blockchair into bitcoinlib/data/providers.examples.json and bitcoinlib/data/providers.json for now. Is this implementation tested? Is importing groestlcoin_hash not needed for base58 checksum? Also we need to add unit tests for GRS on bitcoinlib/tests

We may not need to copy those files to use those services.  If the implementation is the same, then its probably better not to, but we shall see.

File: bitcoinlib/services/groestlcoind.py [hunk 39]
```diff
@@ -74,25 +74,25 @@ def from_config(configfile=None, network='bitcoin'):
...
                 cfn = Path(location, config_fn).expanduser()
                 if cfn.exists():
                     break
         else:
             cfn = Path(BCL_DATA_DIR, 'config', configfile)
         if not cfn or not cfn.is_file():
-            raise ConfigError("Config file %s not found. Please install bitcoin client and specify a path to config "
-                              "file if path is not default. Or place a config file in .bitcoinlib/bitcoin.conf to "
+            raise ConfigError("Config file %s not found. Please install groestlcoin client and specify a path to config "
+                              "file if path is not default. Or place a config file in .groestlcoinlib/groestlcoin.conf to "
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py:95]: "file if path is not default. Or place a config file in .bitcoinlib/groestlcoin.conf to "

File: bitcoinlib/services/groestlcoind.py [hunk 40]
```diff
@@ -74,25 +74,25 @@ def from_config(configfile=None, network='bitcoin'):
...
             network = network.name
         if network == 'testnet':
-            config_fn = 'bitcoin-testnet.conf'
+            config_fn = 'groestlcoin-testnet.conf'
 
         cfn = None
         if not configfile:
-            config_locations = ['~/.bitcoinlib', '~/.bitcoin', '~/Application Data/Bitcoin',
-                                '~/Library/Application Support/Bitcoin']
+            config_locations = ['~/.groestlcoinlib', '~/.groestlcoin', '~/Application Data/Groestlcoin',
```
Comment (gruve-p) [inline bitcoinlib/services/groestlcoind.py:85]: `config_locations = ['~/.bitcoinlib', '~/.groestlcoin', '~/Application Data/Groestlcoin',`

File: bitcoinlib/services/groesltcoinesplora.py [hunk 41]
```diff
@@ -0,0 +1,186 @@
...
+from bitcoinlib.encoding import to_hexstring
+
+PROVIDERNAME = 'groestlcoinesplora'
+# Please note: In the Blockstream API, the first couple of Bitcoin blocks are not correctly indexed,
+# so transactions from these blocks are missing.
+
+_logger = logging.getLogger(__name__)
+
+
+class GroesltcoinEsploraClient(BaseClient):
```
Comment (gruve-p) [inline bitcoinlib/services/groesltcoinesplora.py]: class GroestlcoinEsploraClient(BaseClient):

**Review by gruve-p — CHANGES_REQUESTED** [review]
Please rename bitcoinlib/services/groesltcoinesplora.py to bitcoinlib/services/groestlcoinesplora.py

File: bitcoinlib/data/providers.json [hunk 42]
```diff
@@ -306,5 +306,71 @@
...
+    "network": "groestlcoin_testnet",
+    "client_class": "CryptoID",
+    "provider_coin_id": "grs-test",
+    "url": "https://chainz.cryptoid.info/",
+    "api_key": "",
+    "priority": 10,
+    "denominator": 100000000,
+    "network_overrides": null
   }
 }
```
Comment (gruve-p) [inline bitcoinlib/data/providers.json:376]: },
  "blockchair.groestlcoin": {
    "provider": "blockchair",
    "network": "groestlcoin",
    "client_class": "BlockChairClient",
    "provider_coin_id": "",
    "url": "https://api.blockchair.com/groestlcoin/",
    "api_key": "",
    "priority": 10,
    "denominator": 100000000,
    "network_overrides": null

Comment (gruve-p) [reply bitcoinlib/data/providers.json:376]: please add blockchair for grs also in providers.json

File: bitcoinlib/services/groestlcoind.py (hunk 39 above)
Comment (gruve-p) [reply bitcoinlib/services/groestlcoind.py:95]: this is not resolved yet

File: bitcoinlib/services/groestlcoind.py (hunk 40 above)
Comment (gruve-p) [reply bitcoinlib/services/groestlcoind.py:85]: this is not resolved yet

**Pull request closed without merging**